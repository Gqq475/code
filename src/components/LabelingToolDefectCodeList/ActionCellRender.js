import React from 'react'
import { connect } from 'react-redux'
// import DeleteDialog from './DeleteDialog'
import { Button } from '@blueprintjs/core'
import {
  delLabelingToolImageDefect,
  saveLabelingToolImageDefect,
  updateLabelingToolImageDefect,
  DELETE_DEFECT_MARK_FROM_INDEX
} from 'routes/OJS/modules/OJS'

type Props = {
  data: Object,
  rowIndex: Number,
  deleteDefectFromId: Function,
  deleteDefectFromIndex: Function,
  saveDefect: Function,
  updateDefect: Function
};

export class ActionCellRender extends React.Component {
  props: Props;
  constructor (props) {
    super(props)
    this.onDelete = this.onDelete.bind(this)
    this.onSave = this.onSave.bind(this)
    // this.state = {
    //   isDeleteBtnOpen: false
    // }
  }

  onDelete () {
    if (this.props.data.id) {
      // 刪除後端資料庫的 defect
      this.props.deleteDefectFromId(this.props.data.id.toString())
    } else {
      // 刪除 redux state 裡的 defect list
      this.props.deleteDefectFromIndex(this.props.rowIndex)
    }
    // this.setState({
    //   isDeleteBtnOpen: !this.state.isDeleteBtnOpen
    // })
  }

  onSave () {
    if (this.props.data.id) {
      // 如果有 id => 修改 defect (PUT)
      this.props.updateDefect(this.props.data)
    } else {
      // 如果沒有 id => 儲存 defect (POST)
      this.props.saveDefect(this.props.data, this.props.rowIndex)
    }
  }

  render () {
    return (
      <div>
        <section>
          <Button className={'pt-button pt-minimal pt-icon-trash'} title='Delete' onClick={this.onDelete} />
          <Button className={'pt-button pt-minimal pt-icon-confirm'} title='Save' onClick={this.onSave} />
        </section>
      </div>
    )
  }
}

export default connect(
  (state) => ({}),
  (dispatch) => ({
    deleteDefectFromId: (delId) => dispatch(delLabelingToolImageDefect(delId)),
    deleteDefectFromIndex: (rowIndex) => dispatch({
      type: DELETE_DEFECT_MARK_FROM_INDEX,
      data: {rowIndex}
    }),
    saveDefect: (defectData, rowIndex) => dispatch(saveLabelingToolImageDefect(defectData, rowIndex)),
    updateDefect: (defectData) => dispatch(updateLabelingToolImageDefect(defectData))
  })
)(ActionCellRender)

// export default ActionCellRender
