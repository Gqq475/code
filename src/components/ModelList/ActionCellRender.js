import React from 'react'
import DeleteDialog from './DeleteDialog'
import classes from './ModelList.scss'

type Props = {
  data: Object,
  api: Object
};

export class ActionCellRender extends React.Component {
  props: Props;

  constructor (props) {
    super(props)
    this.onDelete = this.onDelete.bind(this)
    this.state = {
      isDeleteBtnOpen: false
    }
  }

  onDelete () {
    this.setState({
      isDeleteBtnOpen: !this.state.isDeleteBtnOpen
    })
  }

  render () {
    return (
      <div className={classes.cn}>
        <button onClick={this.onDelete}
          className={'pt-button pt-minimal pt-icon-delete'}
          title='Delete' />
        <DeleteDialog
          api={this.props.api}
          id={this.props.data.id}
          isOpen={this.state.isDeleteBtnOpen}
          toggleDialog={this.onDelete} />
      </div>
    )
  }
}

export default ActionCellRender
