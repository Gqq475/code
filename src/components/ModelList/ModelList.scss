.ModelList-container {
  // width: 80%;
  border-radius: 5px;
  padding: 0 0 10px 0;
  position: relative;
  // height: 360px;
  border: 1px solid #0087DC;

  .ag-table{
    height: 330px;
  }

  .cell {
    border-radius: 5px;
    color: #fff;
    background: #9B9B9B;
    display: inline-block;
    width: 100px;
    overflow: hidden;
    font-size: 12px
  }

  .progress {
    width: initial;
    margin-left: 5px;
    padding: 0 5px;
    background: #2887C3;
    border-radius: 0;
  }
  .cn{
    text-align: center;
  }
  .status{
    padding: 5px 10px;
    border-radius: 5px 0 5px 0;
    color: #fff;
    background: #9B9B9B;
    &.status-s{
      background: #7ED321;
    }

    &.status-d{
      background: #D0021B;
    }

    &.status-w{
      background: #F5A623;
    }

    &.status-p{
      background: #4A90E2;
    }
  }
}
