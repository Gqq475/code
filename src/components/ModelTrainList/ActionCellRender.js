import React from 'react'
// import AddDialog from './AddDialog'
import DeleteDialog from './DeleteDialog'
// import PublishDialog from './PublishDialog'
// import ModifyDialog from './ModifyDialog'
import classes from './ModelTrainList.scss'
// import OurToaster from 'components/OurToaster'

type Props = {
  data: Object,
  api: Object
};

export class ActionCellRender extends React.Component {
  props: Props;

  constructor (props) {
    super(props)
    this.onDelete = this.onDelete.bind(this)
    this.state = {
      isDeleteBtnOpen: false
    }
  }

  onDelete () {
    this.setState({
      isDeleteBtnOpen: !this.state.isDeleteBtnOpen
    })
  }

  render () {
    return (
      <div className={classes.cn}>
        <button onClick={this.onDelete}
          className={'pt-button pt-minimal pt-icon-delete'}
          title='Delete' />
        <DeleteDialog
          api={this.props.api}
          id={this.props.data.id}
          isOpen={this.state.isDeleteBtnOpen}
          toggleDialog={this.onDelete} />
      </div>
    )
  }
}

export default ActionCellRender
