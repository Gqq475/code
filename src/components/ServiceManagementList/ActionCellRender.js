import React from 'react'
import FeedImgDialog from './FeedImgDialog'
import DestroySrvDialog from './DestroySrvDialog'
import DeleteDialog from './DeleteDialog'
import classes from './ServiceManagementList.scss'
import OurToaster from 'components/OurToaster'
import { connect } from 'react-redux'
const qca = __QCA__

type Props = {
  value:String,
  data: Object,
  api: Object,
  token: String
};

export class ActionCellRender extends React.Component {
  props: Props;

  constructor (props) {
    super(props)
    this.onSetDataSrc = this.onSetDataSrc.bind(this)
    this.onDestroy = this.onDestroy.bind(this)
    this.onPause = this.onPause.bind(this)
    this.onDelete = this.onDelete.bind(this)
    this.state = {
      isFeedBtnOpen: false,
      isDestroyBtnOpen: false,
      isDeleteBtnOpen: false
    }
  }

  onPause () {
    console.log('执行暂停')
    fetch(`${qca}/ojs/kernel/stopFeed`,
      {
        headers: {
          'Content-Type': 'application/json',
          'Accept': 'application/json',
          'Authorization': this.props.token
        },
        method: 'POST',
        body: JSON.stringify({'kernelId': this.props.data.kernelId.toString()})
      }
    )
    .then(response => response.json())
    .then((data) => {
      if (data.code === 200000) {
        OurToaster.show({message: '暂停成功!'})
      } else {
        OurToaster.show({message: '暂停失败!'})
      }
    })
  }

  onSetDataSrc () {
    this.setState({
      isFeedBtnOpen: !this.state.isFeedBtnOpen
    })
  }

  onDestroy () {
    this.setState({
      isDestroyBtnOpen: !this.state.isDestroyBtnOpen
    })
  }

  onDelete () {
    this.setState({
      isDeleteBtnOpen: !this.state.isDeleteBtnOpen
    })
  }

  render () {
    switch (this.props.value) {
      case 'started':
        // status ready
        return <div className={classes.cn}>
          <button disabled className={'pt-button pt-minimal pt-icon-trash'} title='Delete' />
          <button disabled={false} onClick={this.onSetDataSrc}
            className={'pt-button pt-minimal pt-icon-document-open'}
            title='Feed image' />
          <button disabled={false} onClick={this.onDestroy}
            className={'pt-button pt-minimal pt-icon-delete'}
            title='Destroy' />
          <button disabled className={'pt-button pt-minimal pt-icon-history'} title='Log' />
          <FeedImgDialog id={this.props.data.id} deviceId={this.props.data.deviceId}
            kernelId={this.props.data.kernelId}
            modelId={this.props.data.modelId}
            isOpen={this.state.isFeedBtnOpen} toggleDialog={this.onSetDataSrc} />
          <DestroySrvDialog deviceId={this.props.data.deviceId}
            kernelId={this.props.data.kernelId}
            isOpen={this.state.isDestroyBtnOpen} toggleDialog={this.onDestroy} />
        </div>

      case 'feeding':
        return <div className={classes.cn}>
          <button disabled className={'pt-button pt-minimal pt-icon-trash'} title='Delete' />
          <button disabled={false}
            onClick={this.onPause}
            className={'pt-button pt-minimal pt-icon-pause'}
            title='Pause' />
          <button disabled={false} onClick={this.onDestroy}
            className={'pt-button pt-minimal pt-icon-delete'}
            title='Destroy' />
          <button disabled className={'pt-button pt-minimal pt-icon-history'} title='Log' />
          <FeedImgDialog id={this.props.data.id} deviceId={this.props.data.deviceId}
            kernelId={this.props.data.kernelId}
            modelId={this.props.data.modelId}
            isOpen={this.state.isFeedBtnOpen} toggleDialog={this.onSetDataSrc} />
          <DestroySrvDialog deviceId={this.props.data.deviceId}
            kernelId={this.props.data.kernelId}
            isOpen={this.state.isDestroyBtnOpen} toggleDialog={this.onDestroy} />
        </div>

      case 'none':
      case 'destroyed':
        return <div className={classes.cn}>
          <button disabled={false}
            className={'pt-button pt-minimal pt-icon-trash'}
            title='Delete'
            onClick={this.onDelete} />
          <button disabled onClick={this.onSetDataSrc}
            className={'pt-button pt-minimal pt-icon-document-open'}
            title='Feed image' />
          <button disabled onClick={this.onDestroy}
            className={'pt-button pt-minimal pt-icon-delete'}
            title='Destroy' />
          <button disabled className={'pt-button pt-minimal pt-icon-history'} title='Log' />
          <FeedImgDialog id={this.props.data.id} deviceId={this.props.data.deviceId}
            kernelId={this.props.data.kernelId}
            modelId={this.props.data.modelId}
            isOpen={this.state.isFeedBtnOpen} toggleDialog={this.onSetDataSrc} />
          <DestroySrvDialog deviceId={this.props.data.deviceId}
            kernelId={this.props.data.kernelId}
            isOpen={this.state.isDestroyBtnOpen} toggleDialog={this.onDestroy} />
          <DeleteDialog id={this.props.data.id} kernelId={this.props.data.kernelId}
            isOpen={this.state.isDeleteBtnOpen} toggleDialog={this.onDelete}
            api={this.props.api} />
        </div>

      default:
        return <div className={classes.cn}>
          <button disabled className={'pt-button pt-minimal pt-icon-trash'} title='Delete' />
          <button disabled onClick={this.onSetDataSrc}
            className={'pt-button pt-minimal pt-icon-document-open'}
            title='Feed image' />
          <button disabled onClick={this.onDestroy}
            className={'pt-button pt-minimal pt-icon-delete'}
            title='Destroy' />
          <button disabled className={'pt-button pt-minimal pt-icon-history'} title='Log' />
          <FeedImgDialog id={this.props.data.id} deviceId={this.props.data.deviceId}
            kernelId={this.props.data.kernelId}
            modelId={this.props.data.modelId}
            isOpen={this.state.isFeedBtnOpen} toggleDialog={this.onSetDataSrc} />
          <DestroySrvDialog deviceId={this.props.data.deviceId}
            kernelId={this.props.data.kernelId}
            isOpen={this.state.isDestroyBtnOpen} toggleDialog={this.onDestroy} />
        </div>
    }
  }
}
export default connect(
  (state) => ({
    token: state.auth.token
  }),
  (dispatch, state) => {
    return {
      dispatch
    }
  }
)(ActionCellRender)
