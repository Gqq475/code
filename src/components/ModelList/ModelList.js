import React from 'react'
import classes from './ModelList.scss'
import {AgGridReact} from 'ag-grid-react'
import ActionCellRender from './ActionCellRender'
import Utils from '../../Utils.js'
import { connect } from 'react-redux'

type Props = {
  setDetail: Function,
};

export class ModelList extends React.Component {
  props: Props;
  constructor (props) {
    super(props)
    const columnDefs = [
      {headerName: '模型编号', width: 120, field: 'id', suppressMovable: true, cellClass: classes.cn},
      {headerName: '模型名称', width: 120, field: 'name', suppressMovable: true, cellClass: classes.cn},
      {headerName: '状态', width: 120, field: 'status', suppressMovable: true, cellClass: classes.cn},
      {headerName: '产品', width: 120, field: 'productCode', suppressMovable: true, cellClass: classes.cn},
      {headerName: '站点', width: 120, field: 'operationCode', suppressMovable: true, cellClass: classes.cn},
      {headerName: '线别', width: 120, field: 'lineCode', suppressMovable: true, cellClass: classes.cn},
      {headerName: '机台', width: 120, field: 'eqCode', suppressMovable: true, cellClass: classes.cn},
      {headerName: '创建时间', width: 180, suppressMovable: true, field: 'createTime', cellClass: classes.cn},
      {headerName: '操作',
        suppressMovable: true,
        width: 80,
        field: 'status',
        suppressSorting: true,
        cellRendererFramework: ActionCellRender
      }
    ]
    this.onRowClicked = this.onRowClicked.bind(this)
    this.datasource = this.datasource.bind(this)
    this.data = {}
    this.state = {
      columnDefs
    }
  }

  onRowClicked (e) {
    this.bypassRender = true
    this.props.setDetail(e.data)
  }

  shouldComponentUpdate () {
    if (this.bypassRender) {
      this.bypassRender = false
      return false
    }

    return true
  }

  datasource () {
    let smIns = this
    return {
      getRows: function (params) {
        let pageSize = 10
        let pageNo = params.endRow / pageSize
        let url = Utils.format(smIns.props.data.mvTemplate, [pageNo, pageSize])
        fetch(url, {
          headers: {
            'Content-Type': 'application/json',
            'Accept': 'application/json',
            'Authorization': smIns.props.token
          }
        })
        .then(response => response.json())
        .then((data) => {
          params.successCallback(data.result.models, data.result.totalCount)
        })
      }
    }
  }

  render () {
    return (
      <div className={classes['ModelList-container']}>
        <section className={`${classes['ag-table']} ag-fresh`}>
          <AgGridReact
            enableSorting
            sortingOrder={['desc', 'asc']}
            rowSelection='single'
            rowModelType='pagination'
            paginationPageSize={10}
            columnDefs={this.state.columnDefs}
            onRowClicked={this.onRowClicked}
            datasource={this.datasource()}
          />
        </section>
      </div>
    )
  }
}
export default connect(
  (state) => ({
    token: state.auth.token
  }),
  (dispatch, state) => {
    return {
      dispatch
    }
  }
)(ModelList)
