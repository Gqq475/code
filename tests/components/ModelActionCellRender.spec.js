import React from 'react'
import { ActionCellRender } from 'components/ModelList/ActionCellRender'

describe('(Component) ActionCellRender', () => {
  it('render normally', () => {
    const data = {
      'id': '1'
    }
    const wrapper = shallow(
      <ActionCellRender data={data} />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
