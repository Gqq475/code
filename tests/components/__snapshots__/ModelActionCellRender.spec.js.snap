// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) ActionCellRender render normally 1`] = `
<div
  className="cn"
>
  <button
    className="pt-button pt-minimal pt-icon-delete"
    onClick={[Function]}
    title="Delete"
  />
  <Connect(DeleteDialog)
    id="1"
    isOpen={false}
    toggleDialog={[Function]}
  />
</div>
`;
